src/autocomplete/dataset.js

```diff
@@ -93,7 +93,7 @@ _.mixin(Dataset.prototype, EventEmitter, {
     var renderArgs = [].slice.call(arguments, 2);
     this.$el.empty();
 
-    hasSuggestions = suggestions && suggestions.length;
+    hasSuggestions = suggestions && suggestions.length > 0;
```

Comment:
I think its the following line that negates the result of line 96 - `suggestions && suggestions.length` would evaluate to `true`, which would mean that [`this._isEmpty = !hasSuggestions;`](https://github.com/algolia/autocomplete.js/blob/master/src/autocomplete/dataset.js#L97) would set _isEmpty to `false`